### PR TITLE
fix: use permanent backoff when stacker set not found

### LINKED ISSUE #5092: [signer] request for upcoming reward set can introduce 5s of delay
When signing blocks in a prepare phase, the anchor block for the upcoming cycle is requested by the signer in order to learn the set of next signing keys. However, the anchor block may not yet exist.  The absence of the anchor block leads to the signer spending 5 seconds waiting for data that isn't available in the first place before it proceeds to handle StackerDB messages.

### PR DESCRIPTION
**Pull request by hstove**
- Closes #5092

When the signer fetches the stacker set for a cycle, and the response is "not found" (because there isn't a PoX anchor block yet), a "permanent" error is returned instead of "transient". This will exit the exponential backoff function that would previously retry this request until the max timeout (5 seconds) was reached.

### CODE REVIEW

**Comment by jferrant** [comment]
Will this cause issues in the prepare phase though when it has not yet been calculated? I see that the "tests::signer::v0::reloads_signer_set_in" test is failing now.

**Review by kantai — DISMISSED** [review]
LGTM -- let's just make sure that the CI tests we expect to pass all pass

File: stacks-signer/src/client/stacks_client.rs [hunk 1]
```diff
@@ -581,7 +581,7 @@ impl StacksClient {
                 backoff::Error::permanent(e.into())
             })?;
             if error_data.err_type == GetStackersErrors::NOT_AVAILABLE_ERR_TYPE {
-                Err(backoff::Error::transient(ClientError::NoSortitionOnChain))
+                return Err(backoff::Error::permanent(ClientError::NoSortitionOnChain));
```
Comment (obycode) [inline stacks-signer/src/client/stacks_client.rs:584]:
```suggestion
                Err(backoff::Error::permanent(ClientError::NoSortitionOnChain))
```

**Review by obycode — APPROVED** [review]
This makes sense to me. I think adding in the `return` might cause a clippy warning though.

**Comment by hstove** [comment]
LGTM! I can't review as it's my PR, but thank you @kantai for fixing this!

**Review by obycode — APPROVED** [review]

**Review by kantai — APPROVED** [review]

**Review by jcnelson — APPROVED** [review]

**Comment by kantai** [comment]
Marking as "merge when ready" -- the failing tests all pass locally. Two of these flaky tests (which are already flaky in develop) I think I have fixes for, so will open a PR to fix them soon.

**Pull request merged**